packages/kbn-pm/src/commands/watch.ts

```diff
@@ -0,0 +1,110 @@
+import chalk from 'chalk';
+import { Observable, Subject } from 'rxjs';
+
+import { Project } from '../utils/project';
+import { topologicallyBatchProjects } from '../utils/projects';
+import { parallelizeBatches } from '../utils/parallelize';
+import { Command } from './';
+
+/**
+ * Name of the script in the package/project package.json file to run during `kbn watch`.
+ */
+const watchScriptName = 'kbn:watch';
+
+/**
+ * Number of milliseconds we wait before we fall back to the default watch handler.
+ */
+const defaultHandlerDelay = 3000;
+
+/**
+ * If default watch handler is used, then it's the number of milliseconds we wait for
+ * any build output before we consider watch task ready.
+ */
+const defaultHandlerReadinessTimeout = 2000;
+
+function getWatchHandlers(buildOutput$: Observable<string>) {
+  const typescriptHandler = buildOutput$
+    .first(data => data.includes('$ tsc'))
+    .map(() =>
+      buildOutput$
+        .first(data => data.includes('Compilation complete.'))
+        .mapTo('tsc')
+    );
+
+  const webpackHandler = buildOutput$
+    .first(data => data.includes('$ webpack'))
+    .map(() =>
+      buildOutput$.first(data => data.includes('Chunk Names')).mapTo('webpack')
```

Comment:
Forgot to mention, it's also easy to write one-line webpack plugin that will _always_ emit "marker" we want on build completion. `Chunk Names` won't appear if `stats` is explicitly disabled webpack config. We don't need this now, just fyi:
```js
plugins: [
    { apply: c => c.plugin('done', () => console.log('kbn:ready')) },
],
```